cli/create.ts

```diff
@@ -27,22 +28,26 @@ export async function handleCreate({
   name,
   description,
   workspace,
+  apiToken,
 }: ArgumentsCamelCase<CreateArgs>) {
-  await createPack(manifestFile, codaApiEndpoint, {name, description, workspace});
+  await createPack(manifestFile, codaApiEndpoint, {name, description, workspace}, apiToken);
 }
 
 export async function createPack(
   manifestFile: string,
   codaApiEndpoint: string,
   {name, description, workspace}: {name?: string; description?: string; workspace?: string},
+  apiToken?: string,
 ) {
   const manifestDir = path.dirname(manifestFile);
   const formattedEndpoint = formatEndpoint(codaApiEndpoint);
   // TODO(alan): we probably want to redirect them to the `coda register`
   // flow if they don't have a Coda API token.
-  const apiKey = getApiKey(codaApiEndpoint);
-  if (!apiKey) {
-    printAndExit('Missing API token. Please run `coda register` to register one.');
+  if (!apiToken) {
+    apiToken = getApiKey(codaApiEndpoint);
+    if (!apiToken) {
+      return printAndExit('Missing API token. Please run `coda register` to register one.');
+    }
```

Comment:
Seems fine if you want to move this all to a helper and then these are all just one-line calls to that.

Reply:
What could that call look like? I could turn the `printAndExit()` call into an exception, but then I'd still need to add exception handling, which would eat away at any simplicity gains.

Reply:
I think you can just copy-paste that as-is into a helper. `printAndExit()` is just doing `process.exit(1)`, so just terminates the script immediately and doesn't need any handling.

Something like:

```typescript
function assertApiToken(codaApiEndpoint: string, cliApiToken?: string): string {
  if (cliApiToken) {
    return cliApiToken;
  }
  const apiKey = getApiKey(codaApiEndpoint);
  if (!apiKey) {
    return printAndExit('Missing API token. Please run `coda register` to register one.');
  }
  return apiKey;
}
```

Reply:
Ah, I see now. I've gone ahead and done that, and for some other duplicated code I noticed as well.